src/views/DataSessionsView.vue

```diff
@@ -17,13 +27,9 @@ function deleteSession(id) {
 	showDeleteDialog.value = true
 }
 
-function loadAllSessions() {
-	// Load in the session details from the API
-	fetch('http://127.0.0.1:8000/api/datasessions/', {
-		'headers': {'Authorization': 'Token 123456789abcdefg'}
-	})
-		.then(response => response.json())
-		.then(data => dataSessions.value = data.results)
+async function loadAllSessions() {
+  const data = await fetchApiCall({url: dataSessionsUrl, method: 'GET', headers: authHeaders, failCallback: handleError})
```

Comment:
sorry for the same comment, but again I think this needs to be in a try/catch block and `datasessions.value = data.results` should be the handlesuccess callback

Reply:
@capetillo your comment got me to go back and check the implementation - is there any instance fetchApiCall willraise an error? Looking at the implementation it looks like it handles all exceptions and will not raise any to the caller, so is a try/catch really needed on the fetchApiCall?

Reply:
But I do agree that the one-liner success callbacks are generally fine and you can easily put those inline in the call to fetchApiCall, so we should probably stick to that until we decide if it needs a refactor or not.

Reply:
haven't used call backs much but if its ok then ill write it in as a lambda

Reply:
@mgdaily it isn't necessary, it's just customary in async functions. But yeah, `fetchApiCall` already has a try/catch. My PRs that also use `fetchApiCall` got approved with try/catch so I thought we were implementing what's generally used to avoid going up the chain to handle rejection

Reply:
> @mgdaily it isn't necessary, it's just customary in async functions. But yeah, `fetchApiCall` already has a try/catch. My PRs that also use `fetchApiCall` got approved with try/catch so I thought we were implementing what's generally used to avoid going up the chain to handle rejection

yeah, I'm just having this realization now. they were approved, but I hadn't realized we weren't capable of passing any exceptions back up to the caller. I don't know if there's a JS best practice, but in general if a function can't actually raise an exception then a try/catch block is not strictly necessary

Reply:
it's not necessarily best practice. From what I know, it's just to stop going up the chain but i guess failcallback does that. I WILL NOT LET THIS FEED MY IMPOSTER SYNDROME